test.sh

```diff
@@ -15,15 +15,21 @@ cd tests
 #
 # Until we can run `all.lua` reliably, we just run the tests that we know to
 # run within reasonable time).
+#
+# By default, when we run all.lua test suite, it runs underlying tests with a non-yk lua interpreter.
 
 LUA=../src/lua
 
+non_serialised="api bwcoercion closure code events gengc pm tpack tracegc vararg goto cstack locals"
+
 # Non-serialised compilation tests
-# YKFIXME: The following tests are known to work with non-serialised JIT
-for test in api bwcoercion closure code events \
-    gengc pm tpack tracegc vararg goto cstack locals; do
+# YKFIXME: The following tests are known to work only with non-serialised JIT
+for test in ${non_serialised}; do
     YKD_SERIALISE_COMPILATION=0 ${LUA} -e"_U=true" ${test}.lua
 done
 
-# Serialised compilation tests
-YKD_SERIALISE_COMPILATION=1 ${LUA} -e"_U=true" all.lua
+# YKFIXME: The following tests are known to work with serialised JIT
+serialised="${non_serialised} sort"
```

Comment:
Is this saying that `sort` only works with `YKD_SERIALISE_COMPILATION=1`? If so, do we know why?

Reply:
We don't know why yet, last time we spoke we said these errors seem to be related to the lack of threading support in YK.
sort.lua only works with `YKD_SERIALISE_COMPILATION=1`, details in the referenced issue.

Reply:
We need to better understand why this requires `YKD_SERIALISE_COMPILATION=1` before we merge this. What `YKD_SERIALISE_COMPILATION=1` does is not run yk's compilation in parallel: have we found a bug in yk that `YKD_SERIALISE_COMPILATION=1` masks or ... ?

Reply:
I've seen similar errors in other tests, for example: https://github.com/ykjit/yklua/issues/52, https://github.com/ykjit/yklua/issues/48, https://github.com/ykjit/yklua/issues/46, https://github.com/ykjit/yklua/issues/44.

This PR intention was to add one more test that works to the CI. I didn't intend to solve the non-serialised compilation error here so I would prefer to investigate `YKD_SERIALISE_COMPILATION=1` separately, It is also documented in the referenced issue https://github.com/ykjit/yklua/issues/46

Reply:
I don't think we should include this test in CI, since *something* is broken, and `YKD_SERIALISE_COMPILATION=1` is just masking the issue and giving us a sense of false confidence.

Reply:
ok, got it.